aui.views/src/AUI/Platform/linux/AMessageBoxImpl.cpp

```diff
@@ -18,73 +20,93 @@
-            using namespace declarative;
-            setContents(Vertical {
-                AText::fromString(message),
-                Centered {
-                    [&]() -> _<AView> {
-                        switch (b) {
-                            case AMessageBox::Button::OK: return Button { "OK"_i18n }.clicked(me::onOk);
-                            case AMessageBox::Button::OK_CANCEL: return Horizontal {
-                                Button { "OK"_i18n }.clicked(me::onOk),
-                                Button { "Cancel"_i18n }.clicked(me::onCancel),
-                            };
-                            case AMessageBox::Button::YES_NO: return Horizontal {
-                                Button { "Yes"_i18n }.clicked(me::onYes),
-                                Button { "No"_i18n }.clicked(me::onNo),
-                            };
-                            case AMessageBox::Button::YES_NO_CANCEL: return Horizontal {
-                                Button { "Yes"_i18n }.clicked(me::onYes),
-                                Button { "No"_i18n }.clicked(me::onNo),
-                                Button { "Cancel"_i18n }.clicked(me::onCancel),
-                            };
-                            default: throw AException("invalid AMessageBox::Button");
-                        }
-                    },
-                }
-            });
-        }
-    signals:
-        emits<AMessageBox::ResultButton /* result */> status;
+static void on_notification_closed(NotifyNotification *notification, gpointer user_data) {
+    const auto *data = static_cast<AMessageBoxContext *>(user_data);
+    g_main_loop_quit(data->loop);
+}
 
-    private:
-        void onOk() {
-            emit status(AMessageBox::ResultButton::OK);
-        }
-        void onCancel() {
-            emit status(AMessageBox::ResultButton::CANCEL);
-        }
-        void onYes() {
-            emit status(AMessageBox::ResultButton::YES);
+static void on_action(NotifyNotification *notification, char *action, gpointer user_data) {
+    auto *data = static_cast<AMessageBoxContext *>(user_data);
+
+    if (strcmp(action, "yes") == 0) {
+        data->result = AMessageBox::ResultButton::YES;
+    } else if (strcmp(action, "no") == 0) {
+        data->result = AMessageBox::ResultButton::NO;
+    } else if (strcmp(action, "ok") == 0) {
+        data->result = AMessageBox::ResultButton::OK;
+    } else if (strcmp(action, "cancel") == 0) {
+        data->result = AMessageBox::ResultButton::CANCEL;
+    } else {
+        data->result = AMessageBox::ResultButton::INVALID;
+    }
+
+    g_main_loop_quit(data->loop);
+}
+
+AMessageBox::ResultButton AMessageBox::show(
+    AWindow *parent, const AString &title, const AString &message, AMessageBox::Icon icon, AMessageBox::Button b) {
+    if (!notify_init("AUI_App")) {
```

Comment:
I tried to reimplement the feature using GTK. I feel it may ask to change more stuff and wrap the entire windowing system under GTK since now it's directly tied to X11. Not sure, I'm not a Linux wizard 😅 

I have an issue with the implementation I did. The dialog opens with the correct action buttons but when doing a selection, it freezes and doesn't close, but the AUI app receives the response and continues to run.

Reply:
I also have issues of where to init gtk properly using `gtk_init()`. For now I've edited the `OSDesktop.cpp` file to make a call of `gtk_init` on Linux platforms but since this is in `aui.core` I don't feel it's the best place for that...

Reply:
Okay, I'll take a look later

Reply:
> windowing system under GTK since now it's directly tied to X11

I've been researching on this recently as well; I'm still wondering do I want to use X11/Wayland directly or switch to gtk. This is a subject for dedicated discussion. As for now, I'm still continuing to use the X11 base.

Reply:
Please apply this patch to your PR.
`git apply linux.patch`
[linux.patch.zip](https://github.com/user-attachments/files/18141090/linux.patch.zip)

Key changes:
- managed to `gtk_init` in `aui.views`
- got rid of loop in favour of `gtk_dialog_run`
- added several lines to process gtk's event loop. This is why the window was freezing.

Reply:
Applied, the changes work great!